### DISCUSSION TITLE
**.test.ts file is ignored

### DISCUSSION THREAD

**Discussion by liaoyeguo** [storybookjs/storybook#25054] [Help]
### Summary

We use play function to do component interaction test and generate coverage report. However there are some utils functions we want to run test too. I write test for them in *.test.ts files, but those files are ignored when I run test-storybook --coverage

**Comment by jonniebigodes** [comment]
@liaoyeguo thanks for reaching out to us with your question. We appreciate it 🙏! One thing I wanted to ask you is, are you providing any additional configuration to the addon? If so, can you share your `.storybook/main.js|ts`  with us? If not, can you take a look at the configuration options we have [here](https://storybook.js.org/docs/writing-tests/test-coverage#configure) and see if it helps you?

Looking forward to hearing from you.

Hope you have a great day.

Stay safe

**Reply by pdc-tstream** [reply]
Hi @jonniebigodes - I am having the same problem.

If I run "npx test-storybook" any files that have .test.ts/js are not run.  

However if I run "npx vitest" it does run the test.ts/js files.

Unfortunately, running via "npx vitest" doesn't seem to configure the storybook tests the same way and I get other failures - perhaps they aren't being run in-browser?

I've looked over the docs you referenced and do not see anything specific to being able to run/not-run the normal test.js/ts files.

Given that it runs with vitest and not with test-storybook, does that give any clue as to what may be missing somewhere in my config files?

Thanks!

**Reply by pdc-tstream** [reply]
Hello again @jonniebigodes,

I stripped down the project to the bare minimum to show what I'm running into.

Runs diff.test.ts file:
% npx vitest

Does not run diff.test.ts file:
% npm run test-storybook

You may ask "why not just run 'npx vitest'?" - I am writing Audio  DSP code and need AudioContext, and for some reason that path does not allow me access to flowing audio.  I'm working that angle, too.

In short - if I can get the test-storybook running **/*test.ts/js files, I'd be golden. 

Thanks!

[test-storybook-not-running-tests.zip](https://github.com/user-attachments/files/20201224/test-storybook-not-running-tests.zip)
